### ISSUE TITLE
Sonoff basic r2 works but generate an error during Home Assistant startup, Home Assistant 0.111.4, Sonoff firmware 3.4.0

### ISSUE THREAD

**Issue by andreavaravallo** [AlexxIT/SonoffLAN#169]
As soon as I remove the sonoff section from configuration.yaml the error disappears.

Here's the error:
```
Jun 20 15:48:23 raspberrypi hass[364]: #033[31m2020-06-20 15:48:23 ERROR (MainThread) [homeassistant.core] Error doing job: Exception in callback EventBus.async_listen_once.<locals>.onetime_listener(<Event homeassistant_stop[L]>) at /srv/homeassistant/lib/python3.7/site-packages/homeassistant/core.py:678#033[0m
Jun 20 15:48:23 raspberrypi hass[364]: Traceback (most recent call last):
Jun 20 15:48:23 raspberrypi hass[364]:   File "/usr/lib/python3.7/asyncio/events.py", line 88, in _run
Jun 20 15:48:23 raspberrypi hass[364]:     self._context.run(self._callback, *self._args)
Jun 20 15:48:23 raspberrypi hass[364]:   File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/core.py", line 690, in onetime_listener
Jun 20 15:48:23 raspberrypi hass[364]:     self._hass.async_run_job(listener, event)
Jun 20 15:48:23 raspberrypi hass[364]:   File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/core.py", line 399, in async_run_job
Jun 20 15:48:23 raspberrypi hass[364]:     self.async_add_job(target, *args)
Jun 20 15:48:23 raspberrypi hass[364]:   File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/core.py", line 329, in async_add_job
Jun 20 15:48:23 raspberrypi hass[364]:     task = self.loop.create_task(target(*args))
Jun 20 15:48:23 raspberrypi hass[364]: TypeError: stop() takes 1 positional argument but 2 were given#033[0m
```

My configuration.yaml
```
# Configure a default setup of Home Assistant (frontend, api, etc)
default_config:

# Text to speech
tts:
  - platform: google_translate
    language: 'it'

group: !include groups.yaml
automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

http:
# base_url: ****
# ssl_certificate: ****
# ssl_key: ****

recorder:
  purge_keep_days: 7
  exclude:
    domains:
      - sensor
      - automation
      - script

history:
  exclude:
    domains:
      - automation
      - weblink
      - updater
    entities:
      - sensor.last_boot
      - sensor.date

sensor:
  - platform: command_line
    name: "RaspCPU"
    command: 'cat /sys/class/thermal/thermal_zone0/temp'
    unit_of_measurement: '°C'
    value_template: '{{ value | multiply(0.001) | round(1)}}'

  - platform: yr
    name: Weather
    monitored_conditions:
      - temperature
      - symbol
      - precipitation
      - windSpeed
      - pressure
      - windDirection
      - humidity
      - fog
      - cloudiness
      - lowClouds
      - mediumClouds
      - highClouds
      - dewpointTemperature

sonoff:
  username: ********
  password: ********
  reload: once
```

**Comment by AlexxIT** [comment]
Please show logs https://youtu.be/Lt5fT4N5Pm8

**Comment by andreavaravallo** [comment]
2020-06-22 09:03:08  DEBUG    main          SysInfo: {'version': '0.111.4', 'dev': False, 'hassio': False, 'virtualenv': True, 'python_version': '3.7.3', 'docker': False, 'arch': 'armv7l', 'os_name': 'Linux', 'os_version': '4.19.118-v7l+'}
2020-06-22 09:03:08  DEBUG    main          AUTO mode start
2020-06-22 09:03:11  DEBUG    sonoff_cloud  1 devices loaded from the Cloud Server
2020-06-22 09:03:11  DEBUG    main          1000b7a457 == Init   | {'uiid': 1, 'extra': {'manufacturer': 'SONOFF', 'model': 'BASIC', 'sw_version': 'PSF-B01-GL v3.4.1'}, 'params': {'version': 8, 'sledOnline': 'on', 'switch': 'off', 'fwVersion': '3.4.1', 'rssi': -58, 'staMac': '...', 'startup': 'off', 'init': 1, 'pulse': 'off', 'pulseWidth': 500, 'timers': [{'mId': '...', 'type': 'repeat', 'at': '20 11 * * 0,1,2,3,4,5,6', 'coolkit_timer_type': 'repeat', 'enabled': 1, 'do': {'switch': 'off'}}, {'mId': '...', 'type': 'repeat', 'at': '30 5 * * 1,2,3,4,5', 'coolkit_timer_type': 'repeat', 'enabled': 1, 'do': {'switch': 'on'}}, {'mId': '...', 'type': 'repeat', 'at': '0 6 * * 0,6', 'coolkit_timer_type': 'repeat', 'enabled': 1, 'do': {'switch': 'on'}}, {'mId': '...', 'type': 'repeat', 'at': '0 7 * * 0,1,2,3,4,5,6', 'coolkit_timer_type': 'repeat', 'enabled': 1, 'do': {'switch': 'off'}}, {'mId': '...', 'type': 'repeat', 'at': '25 10 * * 0,1,2,3,4,5,6', 'coolkit_timer_type': 'repeat', 'enabled': 1, 'do': {'switch': 'on'}}], 'cloud': 'online'}}
2020-06-22 09:03:11  DEBUG    sonoff_local  1000b7a457 <= Local1 | {'switch': 'off', 'startup': 'off', 'pulse': 'off', 'sledOnline': 'on', 'pulseWidth': 500, 'rssi': -58} | 9
2020-06-22 09:03:11  DEBUG    sonoff_cloud  Cloud init: {'error': 0, 'apikey': '...', 'config': {'hb': 1, 'hbInterval': 145}, 'sequence': '1592809391356'}
2020-06-22 09:09:39  DEBUG    sonoff_local  1000b7a457 => Local4 | {'switch': 'on'} <= {'seq': 10, 'sequence': '1592809778947', 'error': 0}
2020-06-22 09:09:39  DEBUG    sonoff_cloud  1000b7a457 <= Cloud3 | {'action': 'update', 'deviceid': '1000b7a457', 'apikey': '...', 'userAgent': 'device', 'sequence': '1592809778947', 'ts': 0, 'params': {'switch': 'on'}, 'from': 'device', 'seq': '10'}
2020-06-22 09:09:39  DEBUG    sonoff_local  1000b7a457 <= Local3 | {'switch': 'on', 'startup': 'off', 'pulse': 'off', 'sledOnline': 'on', 'pulseWidth': 500, 'rssi': -58} | 10
2020-06-22 09:09:42  DEBUG    sonoff_local  1000b7a457 => Local4 | {'switch': 'off'} <= {'seq': 11, 'sequence': '1592809782250', 'error': 0}
2020-06-22 09:09:42  DEBUG    sonoff_cloud  1000b7a457 <= Cloud3 | {'action': 'update', 'deviceid': '1000b7a457', 'apikey': '...', 'userAgent': 'device', 'sequence': '1592809782250', 'ts': 0, 'params': {'switch': 'off'}, 'from': 'device', 'seq': '11'}
2020-06-22 09:09:42  DEBUG    sonoff_local  1000b7a457 <= Local3 | {'switch': 'off', 'startup': 'off', 'pulse': 'off', 'sledOnline': 'on', 'pulseWidth': 500, 'rssi': -58} | 11
2020-06-22 09:09:43  DEBUG    sonoff_local  1000b7a457 => Local4 | {'switch': 'on'} <= {'seq': 12, 'sequence': '1592809783208', 'error': 0}
2020-06-22 09:09:43  DEBUG    sonoff_cloud  1000b7a457 <= Cloud3 | {'action': 'update', 'deviceid': '1000b7a457', 'apikey': '...', 'userAgent': 'device', 'sequence': '1592809783208', 'ts': 0, 'params': {'switch': 'on'}, 'from': 'device', 'seq': '12'}
2020-06-22 09:09:43  DEBUG    sonoff_local  1000b7a457 <= Local3 | {'switch': 'on', 'startup': 'off', 'pulse': 'off', 'sledOnline': 'on', 'pulseWidth': 500, 'rssi': -58} | 12
2020-06-22 09:09:44  DEBUG    sonoff_local  1000b7a457 => Local4 | {'switch': 'off'} <= {'seq': 13, 'sequence': '1592809784239', 'error': 0}
2020-06-22 09:09:44  DEBUG    sonoff_local  1000b7a457 <= Local3 | {'switch': 'off', 'startup': 'off', 'pulse': 'off', 'sledOnline': 'on', 'pulseWidth': 500, 'rssi': -58} | 13
2020-06-22 09:09:44  DEBUG    sonoff_cloud  1000b7a457 <= Cloud3 | {'action': 'update', 'deviceid': '1000b7a457', 'apikey': '...', 'userAgent': 'device', 'sequence': '1592809784239', 'ts': 0, 'params': {'switch': 'off'}, 'from': 'device', 'seq': '13'}

**Comment by AlexxIT** [comment]
Do you use latest SonoffLAN? I think I fixed this issue earlier.

**Comment by AlexxIT** [comment]
Reopen issue if problem still actual

**Comment by varhub** [comment]
I'm having exactly the same issue. `TypeError: stop() takes 1 positional argument but 2 were given`. Only happens on shutdown/restart.

Feels like a typo/api change that affects to stop code of sonoff addon. But I should have to review the code.

My setup: docker based homeassistant/home-assistant:stable (2021.1.4) + SonoffLAN v2.3.2

**Comment by AlexxIT** [comment]
Fixed in latest beta

**Comment by varhub** [comment]
Thank you Alex for the update.

ps: for future people like me, fixed at 9e2b42bb1f0ed767f4d044ed0505cdb332d95c46

**Comment by AlexxIT** [comment]
You can install beta version
https://github.com/AlexxIT/SonoffLAN/releases/tag/v2.4b1

You can do it via HACS without problem.

It will be in next release.
